File: sdi3-180.Cli-SOAP-CSharp/sdi3-180.Cli-SOAP-CSharp/UserInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_180.Cli_SOAP_CSharp
{
    class UserInterface
    {
	    public void run() {
		    comandos();
		    do {
			    opciones();
			    if (getNumber() == 0){
				    Console.WriteLine("\nHasta pronto!");
				    return;
			    }
                else if (getNumber() == 1)
				    new ListarUsuarios().execute();
                else if (getNumber() == 2)
				    opcionDeshabilitarUsuario();
                else if (getNumber() == 3)
				    new ListarComentarios().execute();
                else if (getNumber() == 4)
				    opcionEliminarComentario();
		    } while (true);
	    }
	
	    public void comandos(){
		    Console.WriteLine("Comandos:");
		    Console.WriteLine("0, 1, 2, 3, 4");
	    }

        public void opciones()
        {
            Console.WriteLine("\nOpción: ");
            Console.WriteLine("0.Salir");
            Console.WriteLine("1.Listar usuarios del sistema");
            Console.WriteLine("2.Deshabilitar un usuario del sistema");
            Console.WriteLine("3.Listar comentarios y puntuaciones");
            Console.WriteLine("4.Eliminar comentarios y puntuaciones");
        }
	
	    public void opcionDeshabilitarUsuario(){
		    do{
			    Console.WriteLine("\nIntroducir \"0\" para salir de esta opcion");
                Console.WriteLine("Intriducir ID del usuario: ");
			    if(getNumber() == 0)
				    break;
                new DeshabilitarUsuario().execute(getId());
		    }while(true);	
	    }
	
	    public void opcionEliminarComentario(){
		    do{
			    Console.WriteLine("\nIntroducir \"0\" para salir de esta opcion");
			    Console.WriteLine("Intriducir ID del comentario: ");
                if (getNumber() == 0)
				    break;
                new EliminarComentario().execute(getId());
		    }while(true);
	    }

        public int getNumber()
        {
            try
            {
                int number = int.Parse(Console.ReadLine());
                return number;
            }
            catch (Exception)
            {
                Console.WriteLine("Comando desconocido -_-");
                return -1;
            }
        }

        public long getId()
        {
            try
            {
                long number = long.Parse(Console.ReadLine());
                return number;
            }
            catch (Exception)
            {
                Console.WriteLine("Id invalido, tiene que ser numeros");
                return -1;
            }

        }

    }
}

File: sdi3-180.Cli-EJB/src/com/sdi/client/UserInterface.java
package com.sdi.client;

import java.io.*;

import com.sdi.client.accion.DeshabilitarUsuario;
import com.sdi.client.accion.EliminarComentario;
import com.sdi.client.accion.ListarComentarios;
import com.sdi.client.accion.ListarUsuarios;

public class UserInterface {
	
	private static BufferedReader in;

	public void run() throws IOException {
		in = new BufferedReader(new InputStreamReader(System.in));
		comandos();
		do {
			opciones();
			String[] line = in.readLine().split(" ");
			if (line[0].equals("0")){
				System.out.println("\nHasta pronto!");
				return;
			}
			else if (line[0].equals("1"))
				new ListarUsuarios().execute();
			else if (line[0].equals("2"))
				opcionDeshabilitarUsuario();
			else if (line[0].equals("3"))
				new ListarComentarios().execute();
			else if (line[0].equals("4"))
				opcionEliminarComentario();
			else
				System.err.println("Comando desconocido-_-");
		} while (true);
	}
	
	public void comandos(){
		System.out.println("Comandos:");
		System.out.println("0, 1, 2, 3, 4");
	}
	
	public void opciones(){
		System.out.println("\nOpción: ");
		System.out.println("0.Salir");
		System.out.println("1.Listar usuarios del sistema");
		System.out.println("2.Deshabilitar un usuario del sistema");
		System.out.println("3.Listar comentarios y puntuaciones");
		System.out.println("4.Eliminar comentarios y puntuaciones");
		System.out.print(">");
	}
	
	public void opcionDeshabilitarUsuario() throws IOException{
		in = new BufferedReader(new InputStreamReader(System.in));
		do{
			System.out.println("\nIntroducir \"0\" para salir de esta opcion");
			System.out.println("Intriducir ID del usuario: ");
			System.out.print(">");
			String[] line = in.readLine().split(" ");
			if(line[0].equals("0"))
				break;
			try{
				new DeshabilitarUsuario().execute(Long.parseLong(line[0]));
			}catch(Exception e){
				System.err.println("Id del usuario invalido, tiene que ser numeros");
			}
		}while(true);	
	}
	
	public void opcionEliminarComentario() throws IOException{
		in = new BufferedReader(new InputStreamReader(System.in));
		do{
			System.out.println("\nIntroducir \"0\" para salir de esta opcion");
			System.out.println("Intriducir ID del comentario: ");
			System.out.print(">");
			String[] line = in.readLine().split(" ");
			if(line[0].equals("0"))
				break;
			try{
				new EliminarComentario().execute(Long.parseLong(line[0]));
			}catch(Exception e){
				System.err.println("Id del comentario invalido, tiene que ser numeros");
			}
		}while(true);
	}
	
}

